src/components/input/inputNumber/inputNumber.tsx

```diff
@@ -0,0 +1,204 @@
+import classNames from 'classnames';
+import type React from 'react';
+import { useState } from 'react';
+import { Button } from '../../button';
+import { IconType } from '../../icon';
+import { useInputProps, useNumberMask, type IUseNumberMaskProps } from '../hooks';
+import { InputContainer, type IInputComponentProps } from '../inputContainer';
+
+export interface IInputNumberProps extends Omit<IInputComponentProps, 'onChange'> {
+    /**
+     * Optional string appended to the input value.
+     */
+    suffix?: string;
+    /**
+     * @see IUseNumberMaskProps['onChange']
+     */
+    onChange?: IUseNumberMaskProps['onChange'];
+}
+
+export const InputNumber: React.FC<IInputNumberProps> = (props) => {
+    const { suffix, onChange, ...otherProps } = props;
+    const { containerProps, inputProps } = useInputProps(otherProps);
+
+    const { className: containerClassName, isDisabled, ...otherContainerProps } = containerProps;
+    const {
+        max: inputMax,
+        min: inputMin,
+        step: inputStep,
+        value: inputValue,
+        onBlur,
+        onFocus,
+        onKeyDown,
+        className: inputClassName,
+        ...otherInputProps
+    } = inputProps;
+
+    const { step, min, max, value } = parseInputs(inputValue, inputStep, inputMin, inputMax);
+
+    const [isFocused, setIsFocused] = useState(false);
+    const {
+        ref: numberMaskRef,
+        value: maskedValue,
+        unmaskedValue,
+        setValue,
+    } = useNumberMask({
+        min,
+        max,
+        value,
+        onChange,
+    });
+
+    const suffixedValue = maskedValue && suffix ? maskedValue + suffix : maskedValue;
+
+    const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (e.key === 'ArrowUp') {
+            handleIncrement();
+        } else if (e.key === 'ArrowDown') {
+            handleDecrement();
+        }
+
+        onKeyDown?.(e);
+    };
+
+    const handleFocus = (e: React.FocusEvent<HTMLInputElement>) => {
+        setIsFocused(true);
+        onFocus?.(e);
+    };
+
+    const handleBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+        setIsFocused(false);
+        onBlur?.(e);
+    };
+
+    const handleIncrement = () => {
+        const parsedValue = Number(unmaskedValue ?? 0);
+
+        // return the input value if it's bigger than the max
+        if (parsedValue > max) {
+            setValue(parsedValue.toString());
+            return;
+        }
+
+        // increment directly to the minimum if value is less than the minimum
+        if (parsedValue < min) {
+            setValue(min.toString());
+            return;
+        }
```

Comment:
Can the unmasked value be lower or greater than the min / max? Doesn't `react-imask` already restrict this?

Reply:
Yup, it should

Reply:
Okay I see, then we can also remove this check, right?
```
        // increment directly to the minimum if value is less than the minimum
        if (parsedValue < min) {
            setValue(min.toString());
            return;
        }
```

Reply:
Okay I see, then we can also remove this check, right?
```
        // increment directly to the minimum if value is less than the minimum
        if (parsedValue < min) {
            setValue(min.toString());
            return;
        }
```

Reply:
Nope, this is mimicking the behavior of the native HTML5 input `number`. Regardless of the `step` value, as long as the input `value` is not defined, the first increment goes straight to the `min` value

Reply:
Oh I see, makes sense thanks 👌🏻  Can we align the implementation of the `handleDecrement` function to this one so that the implemented logic looks the same?

Reply:
Sure thing 👍🏾